rfc/0006-kubewarden-versioning.md

```diff
@@ -0,0 +1,210 @@
+happen because a new important feature/bugfix has been added,
+because of marketing reasons,...
+
+Let's assume:
+
+* The current version of the Kubewarden stack is `0.5.3`
+* The current version of the helm chart is `v0.10.3`
+
+The new feature is not worth a major bump of the "unique kubewarden stack version",
+hence we will perform a minor update.
+
+We are going to tag `v0.6.0` of all these components and release them:
+
+* Kubewarden Controller
+* Policy Server
+* kwctl
+
+This will happen **regardless** of the changes done to these projects. For
+example, kwctl is going to have a new release `v0.6.0` even if it didn't
+have any commits since `v0.5.3` was tagged.
+
+The Helm charts are updated in this way:
+
+* The version of container images (Policy Server, Kubewarden
+  Controller) are set to be `0.6.0`
+* The chart `version` attribute is set to `v0.10.4` because no significant
+  changed happened
+* The `appVersion` attribute is set to `0.6.0`
+
+### A patch is done to kwctl
+
+A series of minor issues are found inside of kwctl. We decide to
+tag a new release that includes these fixes.
+
+The version of `kwctl` is updated from `0.6.0` to `0.6.1`.
+No other releases are done
+
+### A patch is done to Kubewarden Controller
+
+A minor issue is fixed with Kubewarden Controller. We decide to
+tag a new release that includes the fix.
+
+The following actions are done:
+
+* We tag a new release of kubewarden-controller: `0.6.1`
+* We update the helm chart that makes use of this image:
+  * We update the references of the image to ensure `0.6.1` is being pulled
+  * We update the `version` number of the chart itself to be
+    change from `0.10.4` to `0.10.5`.
+    
+    **Note:** we don't change the `appVersion` attribute
+
+### A major security issue is found inside of Policy Server
+
+A major security issue is found inside of Policy server.
+Changes are done to both policy server and Kubewarden Controller.
+
+We decide to release a new version of the whole stack, because of
+the size of the changes and the impact. We will bump the
+Kubewarden stack from `0.6.0` to `0.7.0`.
+
+We proceed with the following actions:
+
+* Tag a new release of Kubewarden Controller: `0.7.0`
+* Tag a new release of Policy Server: `0.7.0`
+* Tag a new release of kwctl: `0.7.0`. We do that even though
+  no actual change has been done to kwctl
+* helm chart:
+  * Update all the references to the Kubewarden Controller and
+    Policy Server images
+  * Update `version`: this time the changes involve some updates to
+    the helm chart itself. We deem this is worth a minor update.
+    Because of that, the `version` goes from `0.10.5` to `0.11.0`
+  * Update the `appVersion` to be `0.7.0`
+
+### A bug is found inside of one of the helm charts
+
+A small fix is done to one of the existing helm charts.
+
+The version of the Kubewarden stack is **not** going to be changed.
+Only the `version` inside of the patched helm chart is going to be
+bumped.
+
```

Comment:
I would add an example about a new version that introduces a backwards-incompatible change, for kwctl, or for the charts, and see how it plays with the versioning.